Modules/_threadmodule.c

```diff
@@ -1295,7 +1404,8 @@ static int
 do_start_new_thread(thread_module_state* state,
                     PyObject *func, PyObject* args, PyObject* kwargs,
                     int joinable,
-                    PyThread_ident_t* ident, PyThread_handle_t* handle)
+                    PyThread_ident_t* ident, PyThread_handle_t* handle,
+                    _PyEventRc *thread_is_exiting)
```

Comment:
Why not pass a `ThreadHandleObject*` directly? It's already ref-counted, and you could spare the additional ref-counted event since you're merely setting a flag...

Reply:
Edit: ah, just saw the comment about finalization above. Darn!